charts/datahub/templates/_helpers.tpl

```diff
@@ -71,4 +71,44 @@ Return the appropriate apiVersion for cronjob.
 {{- else -}}
 {{- print "batch/v1beta1" -}}
 {{- end -}}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+{{/*
+Create general image registry, name and tag
+*/}}
+{{- define "datahub.image" -}}
+{{- $registry := .Values.global.imageRegistry | default .Values.image.registry -}}
+{{ $registry }}/{{ .Values.image.repository }}:{{ required "Global or specific tag is required" (.Values.image.tag | default .Values.global.datahub.version) -}}
+{{- end -}}
+
+{{/*
+Create image registry, name and tag for elasticsearch setup job
+*/}}
+{{- define "elasticsearchSetupJob.image" -}}
+{{- $registry := .Values.global.imageRegistry | default .Values.elasticsearchSetupJob.image.registry -}}
+{{ $registry }}/{{ .Values.elasticsearchSetupJob.image.repository }}:{{ required "Global or specific tag is required" (.Values.elasticsearchSetupJob.image.tag | default .Values.global.datahub.version) -}}
+{{- end -}}
+
+{{/*
+Create image registry, name and tag for kafka setup job
+*/}}
+{{- define "kafkaSetupJob.image" -}}
+{{- $registry := .Values.global.imageRegistry | default .Values.kafkaSetupJob.image.registry -}}
+{{ $registry }}/{{ .Values.kafkaSetupJob.image.repository }}:{{ required "Global or specific tag is required" (.Values.kafkaSetupJob.image.tag | default .Values.global.datahub.version) -}}
+{{- end -}}
+
+{{/*
+Create image registry, name and tag for mysql setup job
+*/}}
+{{- define "mysqlSetupJob.image" -}}
```

Comment:
No need to duplicate for each image, just have a generic one like:

```
{{/*
Create image registry, name and tag for a datahub component 
*/}}
{{- define "datahub.image" -}}
{{- $registry := .imageRegistry | default .image.registry -}}
{{ $registry }}/{{ .image.repository }}:{{ required "Global or specific tag is required" (.image.tag | default .version) -}}
{{- end -}}
```

and include it as:

```
image: {{ include "datahub.image" (dict "imageRegistry" .Values.global.imageRegistry "version" .Values.global.datahub.version "image" .Values.image) }}
```

Reply:
ah very nice. Did not know this trick :)